tests/test_job.py

```diff
@@ -43,33 +50,17 @@ def read_namespaced_job_success(name, namespace):
                  'succeeded': 1}
     return  MockObject({"status":return_value})
 
+def read_namespaced_job_running(name, namespace):
+    return_value = {'active': None,
+                 'completion_time': None,
+                 'conditions': None,
+                 'failed': None,
+                 'start_time': datetime.datetime(2020, 7, 20, 5, 12, 35, tzinfo=tzutc()),
+                 'succeeded': 1}
+    return  MockObject({"status":return_value})
+
```

Comment:
That is a confusing one. Is that intentional? You are checking for Running and previous code authors assumed that the status is calculated based on `job.status.conditions[0].type == 'Complete' or  `job.status.conditions[0].type == 'Failed'` and specifically the number of failed, active or succeeded pods does not matter. It begs a line of comment, though.

Reply:
With the current logic, its assumed that the pod will be in `running`state if the `job.status.conditions` is `None`. I will add these as comments to the code.